internal/persistence/ent/client/sip.go

```diff
@@ -44,6 +46,16 @@ func (c *client) CreateSIP(ctx context.Context, s *datatypes.SIP) error {
 		q.SetCompletedAt(s.CompletedAt.Time)
 	}
 
+	var uploaderUUID *uuid.UUID
+	if s.UploaderID != nil {
+		u, err := c.ent.User.Query().Where(user.UUID(*s.UploaderID)).Only(ctx)
+		if err != nil {
+			return newDBError(errors.New("create SIP: uploader not found"))
+		}
+		q.SetUser(u)
+		uploaderUUID = &u.UUID
+	}
+
```

Comment:
This should happen in a DB transaction, even if we don't have update or delete methods for users, it's an easy thing to do that could prevent issues in the future.